backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/invalidation/impl/ShipmentScheduleInvalidateRepository.java

```diff
@@ -609,5 +611,25 @@ public void releaseRecomputeMarkerOutOfTrx(final PInstanceId pinstanceId)
 		final List<Object> sqlParams = ImmutableList.of(pinstanceId);
 		return TypedSqlQueryFilter.of(sql, sqlParams);
 	}
+	
+	@Override
+	public final void invalidateShipmentSchedulesFor(@NonNull final IQuery<I_M_ShipmentSchedule> shipmentScheduleQuery)
+	{
+		final int count = shipmentScheduleQuery.insertDirectlyInto(I_M_ShipmentSchedule_Recompute.class)
+				.mapColumn(I_M_ShipmentSchedule_Recompute.COLUMNNAME_M_ShipmentSchedule_ID, 
+						I_M_ShipmentSchedule.COLUMNNAME_M_ShipmentSchedule_ID)
+				
+				.execute()
+				.getRowsInserted();
+
+		logger.debug("Invalidated {} shipment schedules for {}", new Object[] { count, shipmentScheduleQuery });
```

Comment:
for some reason i started the review at the bottom..getting to this point from there, i think you need to fire a cache invalidation event here. that way 
* every client will be notified and can update their view (not just the user who ran the process)
* we (probably) don't need to create FUD by having the invalidateView call in the process

Reply:
done directly in de.metas.inoutcandidate.invalidation.impl.ShipmentScheduleInvalidateRepository.invalidateShipmentSchedulesFor(IQuery<I_M_ShipmentSchedule>)